### PR TITLE
Added support for descriptions

### PR DESCRIPTION
**Pull request by strandloper**
Added the ability to provide descriptions via an overloaded constructor. The Description property returns the provided description, or the name if no description was provided.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -11,21 +11,30 @@ Thanks for [Scott Depouw](https://github.com/sdepouw) for his help with this!
...
+		// This constructor is optional, required only if you need to use descriptions
+        protected TestEnum(string name, int value, string description) : base(name, value, description)
+        {
         }
     }
 ```
 
+### Description
+
+When a description is provided using the optional constructor argument it is returned by the `Description` property. If no description was provided, the `Description` property returns the name. Descriptions useful when you need to provide a list of options to a user and the names are not suffieciently descriptive; for instance in a dropdown list.
```
Comment (ardalis) [inline README.md]: "Descriptions *are* useful"

**Comment by ardalis** [comment]
Thanks for your PR! I'm not sure a Description property is something that *every* SmartEnum would need to take advantage of (even optionally). It seems like it might make more sense as a sample demonstrating how one can easily extend the SmartEnum class to suit one's own needs.

**Comment by strandloper** [comment]
Ideally one should be able to use the standard data annotations to add a description attribute, e.g.

```
[Description("One (default)")]
public static TestEnum One = new TestEnum(nameof(One), 1);
```

Then have a `GetListWithDescriptions()` method in the SmartEnum base class. I tried to figure out how to retrieve the attributes but couldn't. Perhaps it isn't possible with them being static, but I'll play some more. If I can't figure it out I will just add an example to the README to show how you could add the description support to your class without anything in the base.

**Comment by ardalis** [comment]
Closing. Feel free to add an example to the README showing how one would achieve this (without updating the contstructor). Thanks!

**Pull request closed without merging**